src/samples/iio+network/iio-gyroscope-console-and-mqtt-publish.c

```diff
@@ -0,0 +1,542 @@
+    struct iio_gyroscope_data *gyro_data = data;
+    struct sol_direction_vector out;
+
+#ifdef MQTT
+    struct sol_buffer payload;
+    struct sol_mqtt_message mqtt_message;
+#endif
+#ifdef GYRO_CALIBRATE
+    struct gyro_cal_t *cal_data = &(gyro_data->cal_data);
+#endif
+#ifdef DENOISE_MEDIAN
+    struct filter_median_t *filter_data = &(gyro_data->filter_data);
+#endif
+    r = sol_iio_read_channel_value(gyro_data->channel_x, &out.x);
+    SOL_INT_CHECK_GOTO(r, < 0, error);
+
+    r = sol_iio_read_channel_value(gyro_data->channel_y, &out.y);
+    SOL_INT_CHECK_GOTO(r, < 0, error);
+
+    r = sol_iio_read_channel_value(gyro_data->channel_z, &out.z);
+    SOL_INT_CHECK_GOTO(r, < 0, error);
+
+    /*
+     * For noisy sensors drop a few samples to make sure we have at least
+     * GYRO_DROP_SAMPLES events in the filtering queue.
+     * This improves mean and std dev.
+     */
+    if (gyro_data->drop_samples_count < GYRO_DROP_SAMPLES) {
+        gyro_data->drop_samples_count++;
+        return;
+    }
+
+    sol_iio_mount_calibration(device, &out);
+
+#ifdef GYRO_CALIBRATE
+    if (cal_data->calibrated == false)
+        cal_data->calibrated = gyro_collect(cal_data, out.x, out.y, out.z);
+
+    out.x = out.x - cal_data->bias_x;
+    out.y = out.y - cal_data->bias_y;
+    out.z = out.z - cal_data->bias_z;
+#endif
+
+#ifdef DENOISE_MEDIAN
+    denoise_median(filter_data, &out.x, &out.y, &out.z);
+#endif
+
+#ifdef GYRO_CALIBRATE
+    clamp_gyro_readings_to_zero(cal_data, &out.x, &out.y, &out.z);
+#endif
+
+#ifdef GYRO_CALIBRATE
+    snprintf(buffer, sizeof(buffer)-1, "%f\t%f\t%f\t[rad/sec]\t(%s)", out.x, out.y, out.z,
+        (cal_data->calibrated ? "Calibrated" : "Not calibrated"));
+#else
+    snprintf(buffer, sizeof(buffer)-1, "%f\t%f\t%f\t[rad/sec]\t(Calibration disabled)",
+        out.x, out.y, out.z);
+#endif
+    printf("%s\n", buffer);
+
+#ifdef MQTT
+    if (gyro_data->mqtt == NULL) {
+        SOL_WRN("mqtt is null");
+        return;
+    }
+    if (sol_mqtt_get_connection_status(gyro_data->mqtt) == SOL_MQTT_CONNECTED) {
+
+        payload = SOL_BUFFER_INIT_CONST(buffer, strlen(buffer));
+
+        mqtt_message = (struct sol_mqtt_message){
+            SOL_SET_API_VERSION(.api_version = SOL_MQTT_MESSAGE_API_VERSION, )
+            .topic = gyro_data->mqtt_topic,
+            .payload = &payload,
+            .qos = SOL_MQTT_QOS_EXACTLY_ONCE,
+            .retain = false,
+        };
+
+        if (sol_mqtt_publish(gyro_data->mqtt, &mqtt_message)) {
+            SOL_WRN("Unable to publish message");
+        }
+    }
+#endif
+    return;
+
+error:
+    SOL_WRN("Could not read channel buffer values");
+}
+
+int
+main(int argc, char *argv[])
+{
+    struct sol_iio_device *device = NULL;
+    struct sol_iio_config iio_config;
+    struct sol_iio_channel_config channel_config = SOL_IIO_CHANNEL_CONFIG_INIT;
+    struct iio_gyroscope_data gyro_data;
+    int device_id;
+
+    memset(&iio_config, 0, sizeof(struct sol_iio_config));
```

Comment:
minor. you can do it in the declaration.
`struct sol_iio_config iio_config = { };`